tests/test_cli.py

```diff
@@ -694,3 +694,37 @@ def query_server_side_effect(*args, **kwargs):
     ] == snapshot
 
     assert len(bat_calls) == 2
+
+
+def test_server_does_not_exist_error(runner_with_error, mocker):
+    mocker.patch(
+        "seagoat.cli.requests.get", side_effect=requests.exceptions.ConnectionError
+    )
+    query = "JavaScript"
+    result = runner_with_error.invoke(seagoat, [query])
+
+    assert result.exit_code == 3
+    assert "The SeaGOAT server is not running" in result.output
+
+
+@pytest.mark.usefixtures("mock_query_server")
+def test_no_network_to_update(runner_with_error, mocker, mock_response):
+    mocked_response = {
+        "stats": {
+            "chunks": {"analyzed": 100, "unanalyzed": 0},
+            "queue": {"size": 0},
+            "accuracy": {"percentage": 100},
+        },
+        "version": __version__,
+    }
+
+    mocker.patch("requests.get", return_value=mock_response(mocked_response))
+    mocker.patch("seagoat.cli.get_server_info", return_value={"address": ""})
+    mocker.patch(
+        "seagoat.cli.warn_if_update_available",
+        side_effect=requests.exceptions.ConnectionError,
+    )
+    result = runner_with_error.invoke(seagoat, ["search"])
```

Comment:
```suggestion
    result = runner_with_error.invoke(seagoat, ["search", repo.working_dir])
```

That should fix test isolation I think. `repo` is a pytest fixture, you can just include it in the arguments and it should work.

I wonder if there is a way to automatically detect such scenarios :thinking: maybe we could have a pytest fixture that is always applied and mocks this function:

```python


def _get_server_data_file_path() -> Path:
    user_cache_dir = Path(appdirs.user_cache_dir("seagoat-servers"))
    user_cache_dir.mkdir(parents=True, exist_ok=True)
    return user_cache_dir / "serverData.json"
```

in such a way that the mock runs the real implementation, but when the returned folder is not a temp folder it throws an exception saying that the test isolation is broken because the test runs on a non-isolated repo

Reply:
anyways that is not required for merging this

Reply:
Either that or a check in conftest that would abort the tests if the server is already running

Reply:
sth like this 

```python

@pytest.fixture(autouse=True)
def disallow_running_in_real_git_repos(mocker):
    # Could be a dumb way, because it creates an unnecessary temp folder
    def is_temp_directory(path: Path) -> bool:
        temp_dir = Path(tempfile.mktemp()).parent
        return path.startswith(temp_dir)

    def mock_wrapper(*args, **kwargs):
        real_path = _get_server_data_file_path(*args, **kwargs)
        if not is_temp_directory(real_path):
            raise RuntimeError("Test isolation is broken because a test runs seagoat on a non-temporary git repo.")
        return real_path

    mocker.patch("seagoat.utils.server._get_server_data_file_path", mock_wrapper)

```

> Either that or a check in conftest that would abort the tests if the server is already running

that could work too, but it could be annoying if you are trying to use seagoat while developing seagoat, as you would need to stop the server every time before running the tests, no?